logstash-core/api/lib/app/pipeline/stats_command.rb

```diff
@@ -0,0 +1,20 @@
+# encoding: utf-8
+require "app/command"
+
+class LogStash::Api::PipelineStatsCommand < LogStash::Api::Command
+
+  def run
+    #return whatever is comming out of the snapshot event, this obvoiusly
+    #need to be tailored to the right metrics for this command.
+    stats = service.get(:pipeline_stats)
```

Comment:
I like that very much, If we want to move away from the observer this will not impact this part +1.